File: cf/1334/c.py
t = int(input())
for _ in range(t):
    n  = int(input())
    a =[]
    for i in range(n):
        u, v = [int(x) for x in input().strip().split()]
        a.append([u,v])

    if n==1:
        res=a[0]
    else:
        mn = 10**10
        si = None
        for i in range(1,n):
            if a[i][0]>a[i-1][1]:
                diff = a[i-1][1]
            else:
                diff = a[i][0]

            if diff<mn:
                mn = diff
                si = i

        if a[0][0]>a[-1][1]:
            diff = a[-1][1]
        else:
            diff = a[0][0]
            
        if diff<mn:
            mn = diff
            si = 0
        # print(si)
        if si is None:
            res = min(a[i][0] for i in range(n))
        else:
            # res=0
            res=a[si][0]
            ct=1
            prev_i=si
            i = si+1
            
            if i==n:
                i=0

            while ct<n:
                # print(i, prev_i, res)
                res+=max(0,a[i][0]-a[prev_i][1])
                prev_i = i
                i+=1
                if i==n:
                    i=0

                ct+=1
        
        print(res)

File: cf/1334/c.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    int n;
    long long int u, v, res=0;
    cin>>n;
    vector<pair<long long int, long long int>>a;
    for(int i=0;i<n;i++){
        cin>>u>>v;
        a.push_back(make_pair(u,v));
    }

    if(n==1)
        res = a[0].first;
    else{
        int si=-1;
        long long int mn;
        long long int diff= a[0].first-max(0ll,(a[0].first-a[n-1].second));
        mn = diff;
        si=0;

        for(int i=1;i<n;i++){
            diff = a[i].first-max(0ll,a[i].first-a[i-1].second);
            if(diff<mn){
                mn = diff;
                si=i;
            }
        }

        int ct=1;
        res+=a[si].first;
        int prev_i = si;
        int i=si+1;
        if(i==n)
            i=0;

        while(ct<n){
            res+= max(0ll,a[i].first-a[prev_i].second);
            prev_i = i;
            i++;
            if(i==n)
                i=0;
            ct++;
        }
    }
    cout<<res<<"\n";
}

int main() {
	ios_base::sync_with_stdio(false);
    cin.tie(NULL);
	
	int t;
	cin>>t;
	while(t--){
		solve();
	}
	
	return 0;
}
